### PR TITLE
lxd/storage/ceph: Implement GetVolumeUsage

### PR DESCRIPTION
**Pull request by maran**
As discussed in #6953

I was unsure what the best way was to parse the JSON response, I decided to go for dedicated structs, I'm open to other options.

### CODE REVIEW

File: lxd/storage/drivers/driver_ceph_volumes.go [hunk 1]
```diff
@@ -602,9 +602,56 @@ func (d *ceph) UpdateVolume(vol Volume, changedConfig map[string]string) error {
...
+type cephDuInfo struct {
+	Images []cephDuLine `json:"images"`
+}
+
+type cephDuLine struct {
+	Name            string `json:"name"`
+	Snapshot        string `json:"snapshot"`
+	ProvisionedSize int64  `json:"provisioned_size"`
+	UsedSize        int64  `json:"used_size"`
+}
```
Comment (stgraber) [inline lxd/storage/drivers/driver_ceph_volumes.go]: As not used outside of the function, you should put those in the function directly, see `GetResources` in driver_ceph.go for an example.

File: lxd/storage/drivers/driver_ceph_volumes.go [hunk 2]
```diff
@@ -602,9 +602,56 @@ func (d *ceph) UpdateVolume(vol Volume, changedConfig map[string]string) error {
...
+			"--pool", d.config["ceph.osd.pool_name"],
+			d.getRBDVolumeName(vol, "", false, false))
+
+		if err != nil {
+			return -1, err
+		}
+
+		var usedSize int64
+		var result cephDuInfo
+		json.Unmarshal([]byte(jsonInfo), &result)
```
Comment (stgraber) [inline lxd/storage/drivers/driver_ceph_volumes.go]: Missing error handling here

File: lxd/storage/drivers/driver_ceph_volumes.go (hunk 1 above)
Comment (maran) [reply lxd/storage/drivers/driver_ceph_volumes.go]: Didn't even know you could define structs in functions, updated.

File: lxd/storage/drivers/driver_ceph_volumes.go (hunk 2 above)
Comment (maran) [reply lxd/storage/drivers/driver_ceph_volumes.go]: Should be solved now.

File: lxd/storage/drivers/driver_ceph_volumes.go (hunk 1 above)
Comment (tomponline) [reply lxd/storage/drivers/driver_ceph_volumes.go]: You can define constants too :)

**Comment by stgraber** [comment]
Looks good to me, let's have jenkins test it and if green, we'll merge.

**Comment by stgraber** [comment]
jenkins: test this please

**Comment by stgraber** [comment]
lxd/storage/drivers/driver_ceph_volumes.go:644:9: undefined: json

So you're missing an import for `encoding/json`.

**Comment by maran** [comment]
Woops, sorry about that. I actually coded this on my LXD dev server and then manually brought the code over locally by copying the function. Seems I forgot about the new import. I will see if I can setup my environment so I don't have to do this in the future.

**Comment by stgraber** [comment]
jenkins: test this please

**Comment by maran** [comment]
I think I also should have removed the else statement as I return in the if. Is that worth updating?

**Comment by stgraber** [comment]
```
lxd/storage/drivers/driver_ceph_volumes.go:617:9: if block ends with a return statement, so drop this else and outdent its block
Found 1 lint suggestions; failing.
```

So yeah, it seems to be complaining about that.

**Comment by stgraber** [comment]
jenkins: test this please

**Comment by stgraber** [comment]
Thanks!

**Pull request merged**